### PR TITLE
added greensock animation post

### PR DESCRIPTION
**Pull request by jimmykodes**

### CODE REVIEW

File: _posts/2019-12-13-green-sock-animation.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+Something frustrating about how eyes work (well, sensory perception in general) is that our sensitivity to opacity isn't linear. An even transition from 0-100% opacity doesn't _look_ like an even transition. Without going into great detail on why this is (mostly because I'm not qualified to...) suffice it to say that, in order to get what _appears_ to be a linear increase in opacity, we need to implement in exponential increase in opacity.
+
+Consider the following animations:
+<div style="display: flex; flex-direction: row; width: 50%; justify-content: space-around">
+<div style="background-color: #000000; width: 50px; height: 50px" id="lin_fade"></div>
+<div style="background-color: #000000; width: 50px; height: 50px" id="log_fade"></div>
+</div>
+
+While the fade out may look pretty similar, when fading in, the square on the left seems to appear almost instantly, and slowly gain its last bit of opacity, while the square on the right seems to be an even fade. In reality, the left square is a linear transition, while the right square is exponential. i.e. the graphs of their opacity over time look something like:
```
Comment (ckcollab) [inline _posts/2019-12-13-green-sock-animation.md:59]: that's really weird..... huh, neat

File: _posts/2019-12-13-green-sock-animation.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
+---
+title:  "Animation Beyond the Canvas"
+date: 2019-12-13 00:00:00 +0800
+author: jimmy
+---
+
+Canvas animations are great for eye catching displays, but what about animations on buttons? Or anything else that isn't a canvas? You _could_ use native css animations, but those can be cumbersome, especially for complex animations. Instead, we'll be taking a look at the popular animation library GreenSock.
```
Comment (ckcollab) [inline _posts/2019-12-13-green-sock-animation.md]: you link to Greensock later, but in this opening summary would be nice to link Greensock here as well I suppose?

File: _posts/2019-12-13-green-sock-animation.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+That was all it took to get the same functionality as the keyframes. We create a GreenSock timeline (`tl` here) and we then start adding actions to it. We give it a css selector, and then our arguments, which are essentially our destination. We set our css to opacity 0 so that at page load, the elements are hidden, and then GreenSock's `to` functionality handles the tweening to get the element to its destination (opacity of 1). 
+
+Timelines by default are sequential, so the next action doesn't start until the previous one has finished, but we can add modifiers to change this. `"-=1"` is one such modifier. This is telling GreenSock to start this animation 1 second _before_ the previous action finishes, so they overlap. So now how do we fix the fact that the fade doesn't look linear? Simple, we add an ease, which GreenSock already has built in. To add an ease, all we have to do is add it as another argument:
+
+```javascript
+tl.to("#bg-image", {opacity: 1, duration: 3, ease: 'power2.in'})
+tl.to(".buttons", {opacity: 1, duration: 3, ease: 'power2.in'}, "-=1")
+``` 
+And that is it. We have overlapped our fades, and they look like they fade in evenly. Easy.
+
```
Comment (ckcollab) [inline _posts/2019-12-13-green-sock-animation.md:97]: Could add a disclaimer on licensing with Greensock maybe? If you're making a product to sell to multiple people, not a one off, I believe you need licensing:
https://greensock.com/licensing/

File: _posts/2019-12-13-green-sock-animation.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+```html
+<style>
+#bg-image, .buttons {
+    opacity: 0
+}
+</style>
+<script>
+$(function () {
+let tl = gsap.timeline()
+tl.to("#bg-image", {opacity: 1, duration: 3})
```
Comment (ckcollab) [inline _posts/2019-12-13-green-sock-animation.md]: indentation here?

File: _posts/2019-12-13-green-sock-animation.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+```javascript
+tl.to("#bg-image", {opacity: 1, duration: 3, ease: 'power2.in'})
+tl.to(".buttons", {opacity: 1, duration: 3, ease: 'power2.in'}, "-=1")
+``` 
+And that is it. We have overlapped our fades, and they look like they fade in evenly. Easy.
+
+## Conclusion
+
+I have only scratched the surface of GreenSock, both in this blog post, and in my personal use. There are so many great features in the library; I highly recommend playing with it yourself. The next time you need to animate _anything_ on the web, put on a green sock!
```
Comment (ckcollab) [inline _posts/2019-12-13-green-sock-animation.md:100]: woo :) nice post, tyty!

**Review by ckcollab — CHANGES_REQUESTED** [review]

File: _posts/2019-12-13-green-sock-animation.md (hunk 3 above)
Comment (jimmykodes) [reply _posts/2019-12-13-green-sock-animation.md:97]: 🤷‍♂ Kinda feels like a strange addition. I figure, let the greensock website handle telling people that...

**Pull request merged**